### PR TITLE
Use hostnetwork SCC for the operand

### PR DESCRIPTION
**Pull request by alebedev87**
`hostnetwork` SCC in combination with CRIO publishing the profile data on the HTTP endpoint instead of the unix socket is enough for this version of the operator:
```sh
$ oc get nob
NAME      AGE
cluster   8m54s

$ oc -n node-observability-operator get pods
NAME                                                              READY   STATUS    RESTARTS   AGE
node-observability-ds-6bgqt                                       2/2     Running   2          7m46s
node-observability-ds-jk7jk                                       2/2     Running   0          7m46s
node-observability-ds-x6jvl                                       2/2     Running   2          7m46s
node-observability-operator-controller-manager-6cdb9fd486-f8ttk   2/2     Running   0          4m42s

$ oc -n node-observability-operator get pods node-observability-operator-controller-manager-6cdb9fd486-f8ttk -o yaml | yq .metadata.annotations | grep scc
openshift.io/scc: restricted-v2

$ oc -n node-observability-operator get pods node-observability-ds-6bgqt -o yaml | yq .metadata.annotations | grep scc
openshift.io/scc: hostnetwork-v2

$ oc -n node-observability-operator exec node-observability-ds-6bgqt -c node-observability-agent -- curl -L http://localhost:6060/debug/pprof/heap?debug=1 | head -5
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0heap profile: 16: 214272 [5270: 110316472] @ heap/1048576
1: 81920 [1: 81920] @ 0x41440e 0x415e59 0x418ee5 0x91c897 0x1412c27 0x44b683 0x44b5d1 0x44b5d1 0x44b5d1 0x44b5d1 0x44b5d1 0x44b5d1 0x43e346 0x46ef61
#	0x91c896	github.com/gogo/protobuf/proto.RegisterType+0x216	github.com/gogo/protobuf@v1.3.2/proto/properties.go:555
#	0x1412c26	k8s.io/api/networking/v1.init.0+0x1a6			k8s.io/api@v0.23.0/networking/v1/generated.pb.go:738
#	0x44b682	runtime.doInit+0x122					runtime/proc.go:6498
```

This helps to:
- not have the priviledged operand container with hostPath access
- avoid the usage of a dedicated SCC reducing the operator permissions for quite an important resource (SCC)

### CODE REVIEW

**Review by sherine-k — COMMENTED** [review]
/approve
I had just one doubt in the readme to check

File: docs/README.md [hunk 1]
```diff
@@ -213,26 +210,26 @@ status:
   - name: ip-172-31-83-22.ec2.internal
     ip: 172.31.83.22
     port: 8443
-  output: /run/node-observability/50778b44-d1f8-11ec-9d64-0242ac120002
+  output: /run/node-observability/node-observability/50778b44-d1f8-11ec-9d64-0242ac120002
```
[the remark this answers is not in the document]
Comment (josefkarasek) [reply docs/README.md]: I dont think NodeObservabilityRun controller sets. status.output. Right now it's blank.
The purpose of .status.output is something more centralize, rather then container layer file system

File: pkg/operator/controller/nodeobservability/daemonset.go [hunk 2]
```diff
@@ -115,14 +110,11 @@ func (r *NodeObservabilityReconciler) desiredDaemonSet(nodeObs *v1alpha1.NodeObs
...
 							Args: []string{
 								"--tokenFile=/var/run/secrets/kubernetes.io/serviceaccount/token",
-								"--storage=/run",
+								"--storage=/tmp",
 								fmt.Sprintf("--caCertFile=%s%s", kbltCAMountPath, kbltCAMountedFile),
 							},
 							Resources:                corev1.ResourceRequirements{},
 							TerminationMessagePolicy: corev1.TerminationMessageFallbackToLogsOnError,
-							SecurityContext: &corev1.SecurityContext{
-								Privileged: &privileged,
```
Comment (josefkarasek) [inline pkg/operator/controller/nodeobservability/daemonset.go:124]: :+1: :+1: :+1:

File: pkg/operator/controller/nodeobservability/daemonset.go [hunk 3]
```diff
@@ -115,14 +110,11 @@ func (r *NodeObservabilityReconciler) desiredDaemonSet(nodeObs *v1alpha1.NodeObs
 							Command:         []string{"node-observability-agent"},
 							Args: []string{
 								"--tokenFile=/var/run/secrets/kubernetes.io/serviceaccount/token",
-								"--storage=/run",
+								"--storage=/tmp",
```
Comment (josefkarasek) [inline pkg/operator/controller/nodeobservability/daemonset.go:113]: Blocked by image change in node-observability-agent, right?

**Review by sherine-k — APPROVED** [review]
/lgtm

File: pkg/operator/controller/nodeobservability/daemonset.go (hunk 3 above)
Comment (alebedev87) [reply pkg/operator/controller/nodeobservability/daemonset.go:113]: yes

**Pull request closed without merging**